pkg/rehearse/jobs.go

```diff
@@ -158,85 +164,109 @@ func filterJob(source *prowconfig.Presubmit, allowVolumes bool) error {
+		ciopConfigs:  ciopConfigs,
+		profiles:     profiles,
+		prNumber:     prNumber,
+		refs:         refs,
+		loggers:      loggers,
+		allowVolumes: allowVolumes,
+		templateMap:  fillTemplateMap(templates),
+	}
 }
 
-// ConfigureRehearsalJobs filters the jobs that should be rehearsed, then return a list of them re-configured with the
-// ci-operator's configuration inlined.
-func ConfigureRehearsalJobs(toBeRehearsed config.Presubmits, ciopConfigs config.CompoundCiopConfig, prNumber int, loggers Loggers, allowVolumes bool, templates []config.ConfigMapSource, profiles []config.ConfigMapSource) []*prowconfig.Presubmit {
-	var templateMap map[string]string
-	if allowVolumes {
-		templateMap = make(map[string]string, len(templates))
-		for _, t := range templates {
-			templateMap[filepath.Base(t.Filename)] = t.TempCMName("template")
-		}
+func fillTemplateMap(templates []config.ConfigMapSource) map[string]string {
+	templateMap := make(map[string]string, len(templates))
+	for _, t := range templates {
+		templateMap[filepath.Base(t.Filename)] = t.TempCMName("template")
 	}
-	rehearsals := []*prowconfig.Presubmit{}
+	return templateMap
+}
 
-	rehearsalsFiltered := filterJobs(toBeRehearsed, allowVolumes, loggers.Job)
-	for repo, jobs := range rehearsalsFiltered {
+// ConfigurePresubmitRehearsals adds the required configuration for the presubmits to be rehearsed.
+func (jc *JobConfigurer) ConfigurePresubmitRehearsals() []*prowconfig.Presubmit {
+	var rehearsals []*prowconfig.Presubmit
+	for repo, jobs := range jc.presubmits {
 		for _, job := range jobs {
-			jobLogger := loggers.Job.WithFields(logrus.Fields{"target-repo": repo, "target-job": job.Name})
-			rehearsal, err := makeRehearsalPresubmit(&job, repo, prNumber)
+			jobLogger := jc.loggers.Job.WithFields(logrus.Fields{"target-repo": repo, "target-job": job.Name})
+			rehearsal, err := makeRehearsalPresubmit(&job, repo, jc.prNumber)
 			if err != nil {
 				jobLogger.WithError(err).Warn("Failed to make a rehearsal presubmit")
 				continue
 			}
 
-			rehearsal, err = inlineCiOpConfig(rehearsal, repo, ciopConfigs, loggers)
-			if err != nil {
-				jobLogger.WithError(err).Warn("Failed to inline ci-operator-config into rehearsal job")
+			if err := jc.configureJobSpec(rehearsal.Spec, job.Name); err != nil {
+				jobLogger.WithError(err).Warn("Failed to inline ci-operator-config into rehearsal presubmit job")
 				continue
 			}
 
-			if allowVolumes {
-				replaceCMTemplateName(rehearsal.Spec.Containers[0].VolumeMounts, rehearsal.Spec.Volumes, templateMap)
-				replaceClusterProfiles(rehearsal.Spec.Volumes, profiles, loggers.Debug.WithField("name", job.Name))
-			}
-
 			jobLogger.WithField(logRehearsalJob, rehearsal.Name).Info("Created a rehearsal job to be submitted")
 			rehearsals = append(rehearsals, rehearsal)
 		}
 	}
-
 	return rehearsals
 }
 
+func (jc *JobConfigurer) configureJobSpec(spec *v1.PodSpec, jobName string) error {
```

Comment:
Requiring `jobName` parameter that's only used to build a debug logger looks like a code smell to me (the method does not need a name, it needs a logger). I don't think there's any easy fix for this, maybe consider passing the logger directly? We already create a job-specific logger in `ConfigurePresubmitRehearsals`...

Reply:
addressed